vm-virtio/src/vhost_user/fs.rs

```diff
@@ -183,18 +184,34 @@ impl VhostUserMasterReqHandler for SlaveReqHandler {
             let cache_end = self.cache_offset.raw_value() + self.cache_size;
             let efault = libc::EFAULT;
 
-            let offset = gpa
-                .checked_sub(self.cache_offset.raw_value())
-                .ok_or_else(|| io::Error::from_raw_os_error(efault))?;
-            let end = gpa
-                .checked_add(fs.len[i])
-                .ok_or_else(|| io::Error::from_raw_os_error(efault))?;
+            let mut ptr = if gpa >= self.cache_offset.raw_value() && gpa < cache_end {
+                let offset = gpa
+                    .checked_sub(self.cache_offset.raw_value())
+                    .ok_or_else(|| io::Error::from_raw_os_error(efault))?;
+                let end = gpa
+                    .checked_add(fs.len[i])
+                    .ok_or_else(|| io::Error::from_raw_os_error(efault))?;
 
-            if gpa < self.cache_offset.raw_value() || gpa >= cache_end || end >= cache_end {
-                return Err(io::Error::from_raw_os_error(efault));
-            }
+                if gpa < self.cache_offset.raw_value() || gpa >= cache_end || end >= cache_end {
+                    return Err(io::Error::from_raw_os_error(efault));
+                }
+
+                self.mmap_cache_addr + offset
+            } else {
+                use crate::vm_memory::GuestMemory;
+                let memory: &GuestMemoryMmap = &self.mem.memory();
+                match memory.get_host_address(GuestAddress(gpa)) {
```

Comment:
```suggestion
                match self.mem.memory().get_host_address(GuestAddress(gpa)) {
```
Since we never reuse the memory anywhere, you can simplify these 2 lines.